ramls/instance-iteration/iteration-job-params.json

```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Iteration job parameters",
+  "type": "object",
+  "javaType": "org.folio.rest.jaxrs.model.IterationJobParams",
+  "properties":
+  {
+    "eventType":
+    {
+      "description": "Type of events to be published",
+      "type": "string",
+      "default": "ITERATE"
+    },
```

Comment:
What other possible event types are allowed? The DomainEventType class suggests that ITERATE is the only one possible.

If there is only one valid option, why give the client the choice?

If there are more, could they be explicitly documented as part of the schema using an enumeration?

Reply:
Well, ideally eventType value should not be strictly defined, it's supposed to be whatever caller wants it to be. `ITERATE` just serves as a default value, in case the event type is not explicitly specified. That is why the schema doesn't contain any value restrictions, like enum.

But `DomainEvent` class accepts only enum values as event type. I had to define `ITERATE` in `DomainEventType`, otherwise there would be an exception during transformation from String representation to enum value. 
So there isinconsistency between flexible event type in interface definition and its representation in Java code that is hard to overcome.

Reply:
> Well, ideally eventType value should not be strictly defined, it's supposed to be whatever caller wants it to be. 

What is the value of allowing the caller to change the event type?

As I understand it, event types are a way of conveying to a consumer the semantic meaning (what happened, though that isn't strictly true for these messages as they aren't really events) and what information should be provided.

> But `DomainEvent` class accepts only enum values as event type. I had to define `ITERATE` in `DomainEventType`, otherwise there would be an exception during transformation from String representation to enum value.

What will happen if the client provides a different event type?

This suggests to me that it will fail.

> So there is inconsistency between flexible event type in interface definition and its representation in Java code that forces all the values to be defined. This is hard to overcome unless making event type in `DomainEvent` of String type instead of enum, which is also probably not desirable.

Or we stop the client from providing the name of the event type. The type itself doesn't really change, as the same semantic meaning and structure is used whatever name is provided.